sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CountMinSketchAgg.scala

```diff
@@ -208,3 +209,20 @@ case class CountMinSketchAgg(
       confidenceExpression = third,
       seedExpression = fourth)
 }
+
+object CountMinSketchAggExpressionBuilder extends ExpressionBuilder {
+  final val functionSignature = FunctionSignature(Seq(
+    NamedArgument("column",
+                  FixedArgumentType(TypeCollection(IntegralType, StringType, BinaryType))),
+    NamedArgument("epsilon", FixedArgumentType(DoubleType)),
+    NamedArgument("confidence", FixedArgumentType(DoubleType)),
+    NamedArgument("seed", FixedArgumentType(IntegerType))
```

Comment:
We need to make a decision here. Before this PR, SQL functions define the required input types in the corresponding `Expressions`. Now we want to replace it with a new framework. This is fine but we need to make sure the new framework can fully cover the old one.

The old framework is actually quite flexible, you can do arbitrary checks in `Expression.checkInputDataTypes`, or use some convenient trait like `ExpectsInputTypes`. The old framework can also connect to the type coercion framework. For example, the `ImplicitCastInputTypes` trait.

The `PercentileBase` expression is an example. It requires the `percentage` input to be foldable, double-type, and within the range [0, 1.0]. Can the new framework cover it?

Reply:
I don't think the new framework will check the data types or the input requirements - this will still be the responsibility and functionality of original `checkInputDataTypes`. The new framework basically just reorder the arguments based on the **names**. In the current implementation it doesn't really use the type information in the signatures.

Reply:
It's good to make the new framework more capable, but we need to have a clear vision for the eventual system. If eventually we only want the new named-parameter framework to reorder the function parameters, and leave the type check to the existing `Expression` framework, then the named-parameter framework should not define the type information. If we want to move everything to this new framework, we should design it well, by looking at existing functions and define how everything should be done with the new framework.